neuraxle/steps/numpy.py

```diff
@@ -373,3 +374,198 @@ def __init__(self, new_shape):
+    """
+    Compute `np.mean <https://numpy.org/devdocs/reference/generated/numpy.mean.html>`_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a mean.
+
+        :param data_inputs: data inputs
+        :return: data inputs mean for the given axis
+        """
+        return np.mean(data_inputs, axis=self.axis)
+
+
+class NumpyMedian(NonFittableMixin, BaseStep):
+    """
+    Compute `np.median <https://numpy.org/doc/1.18/reference/generated/numpy.median.html>`_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a median.
+
+        :param data_inputs: data inputs
+        :return: data inputs median for the given axis
+        """
+        return np.median(data_inputs, axis=self.axis)
+
+
+class NumpyMin(NonFittableMixin, BaseStep):
+    """
+    Compute `np.min <https://numpy.org/doc/1.18/reference/generated/numpy.minimum.html>`_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a min.
+
+        :param data_inputs: data inputs
+        :return: min value for the given axis
+        """
+        return np.min(data_inputs, axis=self.axis)
+
+
+class NumpyMax(NonFittableMixin, BaseStep):
+    """
+    Compute `np.max <https://numpy.org/doc/1.18/reference/generated/numpy.ndarray.max.html>̀_ at the given axis.
+
+    .. seealso::
+        :class:`~neuraxle.base.BaseStep`,
+        :class:`~neuraxle.base.NonFittableMixin`
+    """
+    def __init__(self, axis=None):
+        BaseStep.__init__(self)
+        NonFittableMixin.__init__(self)
+
+        if axis is None:
+            axis = -2
+        self.axis = axis
+
+    def transform(self, data_inputs):
+        """
+        Will featurize data with a max.
+
+        :param data_inputs: 3D time series of shape [batch_size, time_steps, sensors]
+        :return: max value for the given axis
+        """
+        return np.max(data_inputs, axis=self.axis)
+
+
+class FFTPeakBinWithValue(NonFittableMixin, BaseStep):
```

Comment:
Would be cool to add this as a prebuild class as well: 

```
class Cheap3Dto2DTransformer(ChooseOneOrManyStepsOf):
    def __init__(self):
	super([
            Pipeline([
                NumpyFFT(),
                NumpyAbs(),
                ChooseOneOrManyStepsOf([
                    NumpyFlattenDatum(),  # Reshape from 3D to flat 2D: flattening data except on batch size
                    FFTPeakBinWithValue()  # Extract 2D features from the 3D FFT bins
                ], joiner=NumpyConcatenateInnerFeatures())
            ]),
            NumpyMean(),
            NumpyMedian(),
            NumpyMin(),
            NumpyMax()
        ], joiner=NumpyConcatenateInnerFeatures()),
```

This will be super useful for new pipelines in which we will need to flatten 3D to 2D data for classif!